recipes/libstudxml/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, MSBuild, tools
+import os
+
+class LibStudXmlConan(ConanFile):
+    name = "libstudxml"
+    description = "A streaming XML pull parser and streaming XML serializer implementation for modern, standard C++."
+    topics = ("xml", "xml-parser", "serialization")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.codesynthesis.com/projects/libstudxml/"
+    license = "MIT"
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("expat/2.4.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            args = ["--with-extern-expat", "CXXFLAGS=\"-UNDEBUG\""]
```

Comment:
Massive hack: their tests only build with `assert()` present, this flag makes sure it does build with `build_type=Release`.
I'm working on another version that patches their Makefile.am/configure.ac to stop building tests all together

Reply:
How about adding an `assert` define?
```python
if self.settings.build_type != "Release":
    self._autotools.defines.append("assert=")
```

Reply:
Not really, they check for the NDEBUG macro directly instead of simply assert (tests/parser/driver.cxx):
```cpp
#ifdef NDEBUG
#  error tests require enabled assert(); un-define the NDEBUG macro
#endif
```

Reply:
Fixed by patching their autotools files. This was needed anyways so that when compiling with libc++ (clang 8 specifically), it doesn't end up including `version` from libstudxml (instead of libc++).